util/src/main/java/io/camunda/zeebe/util/health/CriticalComponentsHealthMonitor.java

```diff
@@ -168,17 +176,17 @@ public void onRecovered() {
     }
 
     @Override
-    public void onUnrecoverableFailure() {
-      actor.run(this::onComponentDied);
+    public void onUnrecoverableFailure(final HealthReport report) {
+      actor.run(() -> onComponentDied(report));
     }
 
-    private void onComponentFailure() {
+    private void onComponentFailure(final HealthReport report) {
       if (!monitoredComponents.containsKey(componentName)) {
         return;
       }
 
       log.warn("{} failed, marking it as unhealthy", componentName);
```

Comment:
:x: I think we should also log the health report here. Reason: The other log is "debug" level. It would be good to get these info even if "debug" level is not enabled.